test/access/NonceTracker.test.js

```diff
@@ -0,0 +1,67 @@
+
+import assertRevert from '../helpers/assertRevert';
+
+const NonceTracker = artifacts.require('NonceTrackerImpl');
+
+require('chai')
+  .use(require('chai-as-promised'))
+  .use(require('chai-bignumber')(web3.BigNumber))
+  .should();
+
+contract('NonceTracker', ([_, owner, anyone, another]) => {
+  context('canDoThisOnce', function () {
+    before(async function () {
```

Comment:
I see what you're doing here with the test cases following a sequence to prevent code duplication, but I'd rather them be a bit longer and independent - fragile test suites are always a PITA.